builder/docker/step_set_defaults.go

```diff
@@ -0,0 +1,40 @@
+package docker
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+)
+
+type StepSetDefaults struct{}
+
+func (s *StepSetDefaults) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	driver := state.Get("driver").(*DockerDriver)
+	config := state.Get("config").(*Config)
+
+	// Fetch default CMD and ENTRYPOINT
+	defaultCmd, _ := driver.Cmd(config.Image)
+	defaultEntrypoint, _ := driver.Entrypoint(config.Image)
+
+	// Set defaults if not provided by the user
+	hasCmd, hasEntrypoint := false, false
+	for _, change := range config.Changes {
+		if strings.HasPrefix(change, "CMD") {
+			hasCmd = true
+		} else if strings.HasPrefix(change, "ENTRYPOINT") {
+			hasEntrypoint = true
+		}
+	}
+
+	if !hasCmd && defaultCmd != "" {
+		config.Changes = append(config.Changes, "CMD "+defaultCmd)
+	}
+	if !hasEntrypoint && defaultEntrypoint != "" {
+		config.Changes = append(config.Changes, "ENTRYPOINT "+defaultEntrypoint)
+	}
```

Comment:
This change broke our builds because if both values are defined they are connected. We only defined `ENTRYPOINT` and   got the `CMD` from the base image which then was used as default arguments in our entrypoint.